sherpa/astro/io/pyfits_backend.py

```diff
@@ -1465,3 +1465,100 @@ def set_arrays(filename, args, fields=None, ascii=True, clobber=False):
     hdu = fits.table_to_hdu(tbl)
     hdu.name = 'TABLE'
     hdu.writeto(filename, overwrite=True)
+
+
+def _add_header(hdu, header):
+    """Add the header items to the HDU.
+
+    Parameters
+    ----------
+    hdu : astropy HDU
+    header : list of sherpa.astro.io.xstable.HeaderItem
+    """
+
+    for hdr in header:
+        card = [hdr.name, hdr.value]
+        if hdr.desc is not None or hdr.unit is not None:
+            comment = "" if hdr.unit is None else f"[{hdr.unit}] "
```

Comment:
If hdr.unit is not None, comment is set to the value with an extra space behind it. If hdr.desc is None the extra space is left there. Should the space only be added if you are appending a value from hdr.desc?

Reply:
I'll have to look at the `astropy.io.fits` logic